app/src/main/kotlin/cz/covid19cz/erouska/ui/main/MainActivity.kt

```diff
@@ -116,6 +144,31 @@ class MainActivity :
         customTabHelper.chromePackageName?.let {
             CustomTabsClient.bindCustomTabsService(this, it, customTabsConnection)
         }
+
+        if (intent?.action == Intent.ACTION_RUN) {
+            when (intent.data) {
+                ShortcutsActions.URL_PAUSE -> {
+                    L.d("Received action intent to pause the service")
+
+                    startService(CovidService.pause(this))
+
+                    Handler().postDelayed({
+                        L.d("Hiding the activity after handling of action intent")
+                        finishAfterTransition()
+                    }, 2000);
+                }
+                ShortcutsActions.URL_RESUME -> {
+                    L.d("Received action intent to resume the service")
+
+                    startService(CovidService.resume(this))
+
+                    Handler().postDelayed({
+                        L.d("Hiding the activity after handling of action intent")
+                        finishAfterTransition()
```

Comment:
Have you tested if you exit the activity by pressing back before the timeout ends?

Reply:
Timeout removed